src/util/queryLocations.js

```diff
@@ -0,0 +1,176 @@
+ * @param {int} end The time slot the location closes (in minutes since midnight on Sunday)
+ * @returns true if the location is open, false otherwise
+ */
+function isOpen(start, end) {
+  const weekday = now.weekday === 7 ? 0 : now.weekday;
+  const nowMinutes = toMinutes(weekday, now.hour, now.minute);
+
+  return start <= nowMinutes && nowMinutes <= end;
+}
+
+/**
+ * Gets the next available time slot for a given location
+ * @param {any[]} times List of time slots for a location
+ * @returns The next time slot when the location opens
+ */
+function getNextTimeSlot(times) {
+  const weekday = now.weekday === 7 ? 0 : now.weekday;
+  const nowMinutes = toMinutes(weekday, now.hour, now.minute);
+
+  // Find the first time slot that opens after now
+  const nextTimeSlot = times.find(({ start }) => {
+    return start.rawMinutes >= nowMinutes;
+  });
+
+  if (nextTimeSlot == null) {
+    // End of the week. Return the first time slot instead.
+    return times[0];
+  } else {
+    return nextTimeSlot;
+  }
+}
+
+/**
+ * Return the status message for a dining location, given the current or next available
+ * time slot, and whether or not the location is currently open
+ * @param {{ start, end }} timeSlot The current or next available time slot
+ * @param {boolean} isOpen whether or not the location is currently open
+ * @returns {str} The status message for the location
+ */
+function getStatusMessage(timeSlot, isOpen) {
+  const weekday = now.weekday === 7 ? 0 : now.weekday;
+  const nowMinutes = toMinutes(weekday, now.hour, now.minute);
+
+  const { start, end } = timeSlot;
+  // If open, look at when timeSlot closes. If closed, look at when timeSlot opens.
+  const refTime = isOpen ? end : start;
+
+  // Get difference
+  let diff = isOpen ? refTime.rawMinutes - nowMinutes : refTime.rawMinutes - nowMinutes;
+  if (diff < 0) {
+    diff += WEEK_MINUTES;
+  }
+  const diffMinutes = diff % 60;
+  const diffHours = Math.floor((diff / 60) % 24);
+  const diffDays = Math.floor(diff / (60 * 24));
+
+  // Create time string
+  const { hour, minute } = refTime;
+  const hour12H = hour % 12 === 0 ? 12 : hour % 12;
+  const ampm = hour >= 12 ? "PM" : "AM";
+  const minutePadded = minute < 10 ? `0${minute}` : minute;
+  const time = `${hour12H}:${minutePadded} ${ampm}`;
+
+  const action = isOpen ? "Closes" : "Opens";
+  const day = WEEKDAYS[timeSlot.start.day];
+
+  if (diffDays > 1) {
+    return `${action} in ${diffDays} days (${day} at ${time})`;
+  } else if (diffDays === 1) {
+    return `${action} in a day (tomorrow at ${time})`;
+  } else if (diffDays === 0) {
+    if (diffHours >= 1) {
+      return `${action} in ${diffHours} hours (today at ${time})`;
+    } else {
+      return `${action} in ${diffMinutes} minutes (today at ${time})`;
+    }
+  }
+}
+
+async function queryLocations() {
+  try {
+    // Query locations
+    const { data } = await axios.get(BASE_URL);
```

Comment:
Nit: Now that fetch has gotten so good, I don't see the value in loading a library to do what is now a standard utility.